# HG changeset patch
# User Praveen Kumar Muthuswamy <praveen.muthuswamy@oracle.com>
# Date 1428520414 25200
#      Wed Apr 08 12:13:34 2015 -0700
# Node ID 045f212982ea4578a9dafb925ad62cf366802dc6
# Parent  6470d3cc3ce1677a8283cdf15257c2b8c0ab43c9
/*
 * This patch file (dhcp-dynamiclibraries.patch) was developed in-house to
 * allow isc-dhcp to build and package shared objects for libomapi, libdhcpctl,
 * and libdst. Upstream (ISC) has been made aware of this requirement and the
 * the patch through a service ticket and they are actively working on
 * developing the patch.
 */

diff --git a/client/Makefile.am b/client/Makefile.am
--- a/client/Makefile.am
+++ b/client/Makefile.am
@@ -5,7 +5,7 @@
 		   scripts/netbsd scripts/nextstep scripts/openbsd \
 		   scripts/solaris scripts/openwrt
 dhclient_LDADD = ../common/libdhcp.a ../minires/libres.a \
-		 ../omapip/libomapi.a ../dst/libdst.a
+		 ../omapip/libomapi.la ../dst/libdst.la
 man_MANS = dhclient.8 dhclient-script.8 dhclient.conf.5 dhclient.leases.5
 EXTRA_DIST = $(man_MANS)
 
diff --git a/common/tests/Makefile.am b/common/tests/Makefile.am
--- a/common/tests/Makefile.am
+++ b/common/tests/Makefile.am
@@ -13,7 +13,7 @@
 alloc_unittest_SOURCES = test_alloc.c $(top_srcdir)/tests/t_api_dhcp.c
 alloc_unittest_LDADD = $(ATF_LDFLAGS)
 alloc_unittest_LDADD += ../libdhcp.a  \
-	../../omapip/libomapi.a ../../dst/libdst.a ../../minires/libres.a
+	../../omapip/libomapi.la ../../dst/libdst.la ../../minires/libres.a
 
 check: $(ATF_TESTS)
 	atf-run | atf-report
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -37,7 +37,7 @@
 # Use this to define _GNU_SOURCE to pull in the IPv6 Advanced Socket API.
 AC_USE_SYSTEM_EXTENSIONS
 
-AC_PROG_RANLIB
+AC_PROG_LIBTOOL
 AC_CONFIG_HEADERS([includes/config.h])
 
 # we sometimes need to know byte order for building packets
diff --git a/dhcpctl/Makefile.am b/dhcpctl/Makefile.am
--- a/dhcpctl/Makefile.am
+++ b/dhcpctl/Makefile.am
@@ -1,15 +1,16 @@
 bin_PROGRAMS = omshell
-lib_LIBRARIES = libdhcpctl.a
+lib_LTLIBRARIES = libdhcpctl.la
+libdhcpctl_la_LDFLAGS = -version-info 1
 noinst_PROGRAMS = cltest
 man_MANS = omshell.1 dhcpctl.3
 EXTRA_DIST = $(man_MANS)
 
 omshell_SOURCES = omshell.c
-omshell_LDADD = libdhcpctl.a ../common/libdhcp.a ../minires/libres.a \
-                ../omapip/libomapi.a ../dst/libdst.a
+omshell_LDADD = libdhcpctl.la ../common/libdhcp.a ../minires/libres.a \
+                ../omapip/libomapi.la ../dst/libdst.la
 
-libdhcpctl_a_SOURCES = dhcpctl.c callback.c remote.c
+libdhcpctl_la_SOURCES = dhcpctl.c callback.c remote.c
 
 cltest_SOURCES = cltest.c
-cltest_LDADD = libdhcpctl.a ../common/libdhcp.a ../minires/libres.a \
-               ../omapip/libomapi.a ../dst/libdst.a
+cltest_LDADD = libdhcpctl.la ../common/libdhcp.a ../minires/libres.a \
+               ../omapip/libomapi.la ../dst/libdst.la
diff --git a/dst/Makefile.am b/dst/Makefile.am
--- a/dst/Makefile.am
+++ b/dst/Makefile.am
@@ -1,8 +1,9 @@
 AM_CPPFLAGS = -DMINIRES_LIB -DHMAC_MD5
 
-lib_LIBRARIES = libdst.a
+lib_LTLIBRARIES = libdst.la
+libdst_la_LDFLAGS = -version-info 1
 
-libdst_a_SOURCES = dst_support.c dst_api.c hmac_link.c md5_dgst.c \
+libdst_la_SOURCES = dst_support.c dst_api.c hmac_link.c md5_dgst.c \
 		   base64.c prandom.c
 
 EXTRA_DIST = dst_internal.h md5.h md5_locl.h
diff --git a/omapip/Makefile.am b/omapip/Makefile.am
--- a/omapip/Makefile.am
+++ b/omapip/Makefile.am
@@ -1,7 +1,8 @@
-lib_LIBRARIES = libomapi.a
+lib_LTLIBRARIES = libomapi.la
+libomapi_la_LDFLAGS = -version-info 1
 noinst_PROGRAMS = svtest
 
-libomapi_a_SOURCES = protocol.c buffer.c alloc.c result.c connection.c \
+libomapi_la_SOURCES = protocol.c buffer.c alloc.c result.c connection.c \
 		     errwarn.c listener.c dispatch.c generic.c support.c \
 		     handle.c message.c convert.c hash.c auth.c inet_addr.c \
 		     array.c trace.c mrtrace.c toisc.c iscprint.c
@@ -9,5 +10,5 @@
 EXTRA_DIST = $(man_MANS)
 
 svtest_SOURCES = test.c
-svtest_LDADD = libomapi.a ../dst/libdst.a
+svtest_LDADD = libomapi.la ../dst/libdst.la
 
diff --git a/relay/Makefile.am b/relay/Makefile.am
--- a/relay/Makefile.am
+++ b/relay/Makefile.am
@@ -2,7 +2,7 @@
 
 sbin_PROGRAMS = dhcrelay
 dhcrelay_SOURCES = dhcrelay.c
-dhcrelay_LDADD = ../common/libdhcp.a ../omapip/libomapi.a ../dst/libdst.a ../minires/libres.a
+dhcrelay_LDADD = ../common/libdhcp.a ../omapip/libomapi.la ../dst/libdst.la ../minires/libres.a
 man_MANS = dhcrelay.8
 EXTRA_DIST = $(man_MANS)
 
diff --git a/server/Makefile.am b/server/Makefile.am
--- a/server/Makefile.am
+++ b/server/Makefile.am
@@ -13,9 +13,9 @@
 		dhcpv6.c mdb6.c
 
 # libomapi.a this is here twice to handle circular library dependencies :(
-dhcpd_LDADD = ../common/libdhcp.a ../omapip/libomapi.a ../dst/libdst.a \
-	      ../dhcpctl/libdhcpctl.a ../minires/libres.a \
-	      ../omapip/libomapi.a
+dhcpd_LDADD = ../common/libdhcp.a ../omapip/libomapi.la ../dst/libdst.la \
+	      ../dhcpctl/libdhcpctl.la ../minires/libres.a \
+	      ../omapip/libomapi.la
 
 man_MANS = dhcpd.8 dhcpd.conf.5 dhcpd.leases.5
 EXTRA_DIST = $(man_MANS)
diff --git a/server/tests/Makefile.am b/server/tests/Makefile.am
--- a/server/tests/Makefile.am
+++ b/server/tests/Makefile.am
@@ -17,8 +17,8 @@
           ../ddns.c ../dhcpleasequery.c ../dhcpv6.c ../mdb6.c        \
           ../dhcpd.c
 
-DHCPLIBS = $(top_builddir)/common/libdhcp.a $(top_builddir)/omapip/libomapi.a    \
-          $(top_builddir)/dhcpctl/libdhcpctl.a $(top_builddir)/dst/libdst.a \
+DHCPLIBS = $(top_builddir)/common/libdhcp.a $(top_builddir)/omapip/libomapi.la    \
+          $(top_builddir)/dhcpctl/libdhcpctl.la $(top_builddir)/dst/libdst.la \
           $(top_builddir)/minires/libres.a
 
 ATF_TESTS =
